models/v1alpha3/relationship/relationship_helper.go

```
@@ -1,84 +1,161 @@
+// RelationshipDefinition so the identity cannot silently drift with the
+// entity's wire representation: ModelId is json:"-" on the entity, which
+// would drop it from a Marshal of the entity and let equivalent relationships
+// from different models collide on one ID. Every field here is always
+// serialized, so field order and presence are fixed by this struct alone.
+type relationshipIdentity struct {
+	SchemaVersion    core.VersionString         `json:"schemaVersion"`
+	Version          core.SemverString          `json:"version"`
+	Kind             RelationshipDefinitionKind `json:"kind"`
+	RelationshipType string                     `json:"type"`
+	SubType          string                     `json:"subType"`
+	ModelId          *core.Uuid                 `json:"modelId"`
```

golangci-lint (2.13.2)

[error] ST1003: struct field ModelId should be ModelID

(staticcheck)